de.dlr.sc.virsat.graphiti.ui/src/de/dlr/sc/virsat/graphiti/ui/diagram/feature/VirSatChangeColorFeature.java

```diff
@@ -65,22 +69,53 @@ public void execute(ICustomContext context) {
 		s.setLocation(context.getX(), context.getY());
 		ColorDialog colorDialog = new ColorDialog(s);
 		RGB rgbColor = colorDialog.open();
-		if (colorDialog != null && rgbColor != null) {
-			IColorConstant color =
+		if (rgbColor != null) {
+			IColorConstant colorConstant =
 					new ColorConstant(rgbColor.red, rgbColor.green, rgbColor.blue);
+			Color color = manageColor(colorConstant);
 			PictogramElement[] pes = context.getPictogramElements();
 			if (pes != null && pes.length == 1) {
 				for (PictogramElement pe : pes) {
-					GraphicsAlgorithm ga = pe.getGraphicsAlgorithm();
-					ga.setBackground(manageColor(color));
-					for (GraphicsAlgorithm childGa : ga.getGraphicsAlgorithmChildren()) {
-						if (childGa instanceof RoundedRectangle || childGa instanceof Rectangle) {
-							childGa.setBackground(manageColor(color));
-						}
+					if (pe instanceof Connection) {
+						changeColorConnection((Connection) pe, color);
+					} else {
+						changeColorGenericPictorgramElement(pe, color);
 					}
 				}
 			}
 		}	
 	}
+	
+	/**
+	 * Changes the color of a generic pictorgram element
+	 * @param pe the pictogram element
+	 * @param color the target color
+	 */
+	private void changeColorGenericPictorgramElement(PictogramElement pe, Color color) {

```

Comment:
```suggestion
	private void changeColorGenericPictogramElement(PictogramElement pe, Color color) {

```
Guess this change will require some further refactoring...

Reply:
I now also changed it to `changeGenericPictogramElementColor` to make it fit with the other suggestion.